### ISSUE TITLE
Number of scripts: 0

### ISSUE THREAD

**Issue by jesus2099** [OpenUserJs/OpenUserJS.org#729]
All ([my](https://openuserjs.org/users/jesus2099) but [not only](https://openuserjs.org/users/marti)) scripts seem to have recently vanish.

**Comment by Martii** [comment]
Give me a few to investigate this.

**Comment by Martii** [comment]
I've messaged @sizzlemctwizzle  ... I rolled back the commit head on production and it's still only showing minimal pages... he'll have to check out S3 storage which I don't have access to.

**Comment by Martii** [comment]
@sizzlemctwizzle 
dev currently still has it's full compliment ...

everyone else... still awaiting to hear back... this might take some time if he's not available. Sorry.

**Comment by Martii** [comment]
@sizzlemctwizzle 
Nada... downgrading the deps still isn't showing full compliment on local pro and actual production.

**Comment by Martii** [comment]
Arggh... so assigning to @sizzlemctwizzle ... I don't know how to fix this... sizzle mentioned that we might drop S3 and move it to production to avoid these intermittent S3 issues.

**Comment by Martii** [comment]
Added blocking since this is blocking... may modify CONTRIBUTING.md to allow Active Maintainer to add for ESR type situations.

**Comment by Martii** [comment]
OH NO!!!!!! :(

@sizzlemctwizzle 

Ref: Automattic/mongoose#3326

**Comment by Martii** [comment]
sizzle said that my Hello, World! Unit Test script is present on S3 *(I publish that usually after every dep update)* sooooooooooo... that probably means that we still have the source at least just no linkage to it yet.

**Comment by sizzlemctwizzle** [comment]
The source is still there but the DB is devestated. I have a backup that's about a month old. I forgot to turn on automatic backups. I won't make that mistake again.

**Comment by Martii** [comment]
Well a month ago is better than starting from scratch I think. I'd like to keep the published dates intact since that is in the `_id` currently for scripts. We'll lose the users added since then as well... they can always resign up though... I'm feeling awful right now... I've added remove *(delete)* ops to my normal test routines as well.

We'll have to keep this blocking until I remigrate #285 on that DB ... and checkout an older HEAD here.

**Comment by sizzlemctwizzle** [comment]
It just makes me lose so much faith in mongoose. I mean, who in the hell
can get it that wrong? A method that is supposed to remove one document
from a collection removes ALL of the documents in that collection. It's
just ridiculous.

**Comment by chacham2** [comment]
May i help with any SQL related issues?

>I forgot to turn on automatic backups. I won't make that mistake again.
And don't forget to test them once in a while too. Those are 2 mistakes people don;t realize until _after_ they needed them. :)

BTW, oujs is awesome. I hope you get this all cleaned up!

**Comment by jscher2000** [comment]
Would it be useful to post a list of scripts that don't have any metadata page, which makes them difficult to find? I haven't noticed that the site exposes any numeric index for simple spidering.

**Comment by Martii** [comment]
@chacham2 
> May i help with any SQL related issues?

[Contributing](https://github.com/OpenUserJs/OpenUserJS.org/blob/master/CONTRIBUTING.md) is always welcome... I do have my own opinions on the direction as well as being the Active Maintainer... a lot of my thoughts on direction just changed with this issue too... some didn't.

 One thing though is we don't use SQL... it's *mongoose* or MongoDB.

> And don't forget to test them once in a while too.

That is sound advice. Thank you.

@jscher2000 
> Would it be useful to post a list of scripts that don't have any metadata page, which makes them difficult to find?

I presume these exist and those authors need to come forward and let me know what to look for. I can probably, eventually, be able to look it up... but I need their `@name` and their username to do this... without this they are orphaned on S3... this is what sizzle means by the "DB is devastated"... we lost all "linkage" due to that dependency deleting the entire collection instead of a single document *(record)*. As I pointed out to @sizzlemctwizzle this could have been horrifically worse had we had a "delete user" option, as suggested in #359, and someone deleted themselves and \*poof\* all the users would have gone bye bye including myself. :\

> I haven't noticed that the site exposes any numeric index for simple spidering.

SEO is handled differently than GreasyFork *(GF)* and we use username/scriptname in the route. I'm not real fond of looking at GF links as most of the time they are completely useless by not telling you anything until you visit that site. I'm open to improving SEO rankings but that would be another issue  *(and milestone at [SEO](https://github.com/OpenUserJs/OpenUserJS.org/milestones/SEO))* . I seriously doubt a plus vote will happen from me on changing this gently suggested routing url.

@sizzlemctwizzle 
I still can't launch local pro possibly due to what you said at https://github.com/OpenUserJs/OpenUserJS.org/pull/712#issuecomment-131883945 . Since the DB in use is 21ish days old it is pre this comment. I need this access to migrate the DB over the hump to get to 0.2.0 of our project and any manual look ups for those authors that may need their code from S3 slipped in somewhere *(probably not the site itself)*... so far no one has made any requests yet.

**Comment by chacham2** [comment]
@Martii 
I'm not that interested in taking a role, per se, at least not at this point. If SQL is ever required, desired, or seems like it might be nice, i'm willing to help this excellent site.

**Comment by Martii** [comment]
> not at this point.

We'll be here when you are ready. It's okay to take part in issues as well... it is helpful to get a different perspective on what everyone wants versus what can be delivered. OUJS is, in part, about safer choices instead of closed source and/or forcing unreasonable .user.js engine features out of the way like some sites do.

> If SQL is ever required

I'm open to examining this at the *node* level myself... *mongoose* is a different type of DB than I'm normally used to *(I use "squeekel" myself for some applications ;)*... if the benefits outweigh the effort, OUJS as a whole will surely consider it.

**Comment by Martii** [comment]
@sizzlemctwizzle 
Please confirm we have a working back up of the re-migration later this weekend, unblock this issue, and close at your leisure. TIA

**Comment by sizzlemctwizzle** [comment]
Will do.

**Comment by sizzlemctwizzle** [comment]
Confirmed that we have one backup per day for the last 7 days. It is fairly easy to restore from one of these backups.

**Comment by Martii** [comment]
Gratsi! :)
